crates/autopilot/src/infra/solvers/authenticator.rs

```diff
@@ -0,0 +1,91 @@
+use {
+    crate::{
+        domain::{self, eth},
+        infra::blockchain::{
+            contracts::{deployment_address, Contracts},
+            ChainId,
+        },
+    },
+    ethcontract::{dyns::DynWeb3, GasPrice},
+};
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Authenticator {
+    /// The authenticator contract that decides which solver is allowed to
+    /// submit settlements.
+    authenticator: contracts::GPv2AllowListAuthentication,
```

Comment:
Always suspicious when a struct has a field with the same name as the struct itself. The `authenticator` has the concept of a manager which is allowed to add/remove solvers. Maybe `AuthenticatorManager` would be more appropriate.
Or alternatively just `Manager` and then it could be referenced as `authenticator::Manager` from the outside.